test/common/tracing/zipkin/span_buffer_test.cc

```diff
@@ -0,0 +1,100 @@
+#include "common/tracing/zipkin/span_buffer.h"
+
+#include "gtest/gtest.h"
+
+namespace Zipkin {
+
+TEST(ZipkinSpanBufferTest, defaultConstructorEndToEnd) {
+  SpanBuffer buffer;
+  Span span;
+
+  EXPECT_EQ(0ULL, buffer.pendingSpans());
+  EXPECT_EQ("[]", buffer.toStringifiedJsonArray());
+
+  buffer.allocateBuffer(2);
+  EXPECT_EQ(0ULL, buffer.pendingSpans());
+  EXPECT_EQ("[]", buffer.toStringifiedJsonArray());
+
+  buffer.addSpan(std::move(span));
+  EXPECT_EQ(1ULL, buffer.pendingSpans());
+  std::string expected_json_array_string = ""
```

Comment:
you can use R"" as a raw string and get rid of \ inside of the string
that should be more readable